src/companion.jl

```diff
@@ -15,86 +16,78 @@ Also, directly from a polynomial:
 ```julia
 julia> using Polynomials
 
-julia> P=Polynomial([2.0,3,4,5])
+julia> P = Polynomial(2:5)
 Polynomial(2 + 3x + 4x^2 + 5x^3)
 
-julia> C=Companion(P)
+julia> C = Companion(P)
 3×3 Companion{Float64}:
  0.0  0.0  -0.4
  1.0  0.0  -0.6
  0.0  1.0  -0.8
 ```
 """
-struct Companion{T} <: AbstractArray{T, 2}
+struct Companion{T} <: AbstractMatrix{T}
     c :: Vector{T}
 end
 
-# Generate companion matrix from a polynomial
+Companion(v::AbstractVector) = Companion(collect(v))
 
-function Companion(P::Polynomial{T}) where T
-   n = length(P)
-   c = Array{T}(undef, n-1)
-   d=P.coeffs[n]
-   for i=1:n-1
-       c[i]=P.coeffs[i]/d
-   end
-   Companion(c)
+# Construct companion matrix from a polynomial
+
+function Companion(P::Polynomial)
+    n = length(P)
+    c = P.coeffs[1:(n-1)] / P.coeffs[n]
```

Comment:
```suggestion
    c = P.coeffs[begin:end-1] ./ P.coeffs[end]
```

Reply:
Good point about 1-based indexing.  I had overlooked that.
I wish Julia had a subtype of AbstractArray with 1-based indexing.
I will add `Base.require_one_based_indexing` where needed.
I prefer not to add a dependence on `OffsetArrays.jl` to keep this package light.
If you think there is a use case that needs it, let me know.
My goal was simply to make it so that users can use `1:3` instead of having to `collect(1:3)`,
which already adds convenience, not to handle offset arrays.
Someone else can add offsets later if they need it.

Reply:
OffsetArrays.jl can be an "extra" dependency that is only loaded during testing. It will not be a normal dependency.

See https://pkgdocs.julialang.org/v1/api/#Pkg.test

Reply:
Ahh, I see what you mean now.  I had thought you wanted a specialized method for OffsetArrays.  I have updated the test to include a test of an OffsetArray case.
In the process, I updated the .toml files to use the post Julia v1.2 way that uses test/Project.toml
https://pkgdocs.julialang.org/v1/creating-packages/#Adding-tests-to-the-package